libgobuster/libgobuster.go

```diff
@@ -111,10 +112,19 @@ func (g *Gobuster) ClearProgress() {
 // GetRequest issues a GET request to the target and returns
 // the status code, length and an error
 func (g *Gobuster) GetRequest(url string) (*int, *int64, error) {
-	g.incrementRequests()
 	return g.http.makeRequest(url, g.Opts.Cookies)
 }
 
+// DNSLookup looks up a domain via system default DNS servers
+func (g *Gobuster) DNSLookup(domain string) ([]string, error) {
+	return net.LookupHost(domain)
+}
+
+// DNSLookupCname looks up a CNAME record via system default DNS servers
+func (g *Gobuster) DNSLookupCname(domain string) (string, error) {
+	return net.LookupCNAME(domain)
+}
+
```

Comment:
I think I'd prefer to see both of these things go down into the individual plugins. Right now the plugins call out to these three functions (`GetRequest`, `DNSLookup` and `DNSLookupCname`), yet those functions either call default Golang libs or call back into the plugin. Are you open to change this so that these functions disappear?

Reply:
Nevermind. Let's do this as part of a different piece of work.